KsApi/models/ManagePledgeViewBackingEnvelopeTests.swift

```diff
@@ -0,0 +1,258 @@
+
+      XCTAssertEqual(value.project.id, "UHJvamVjdC00NDc0NzM2MTM=")
+      XCTAssertEqual(value.project.name, "The Keyboardio Atreus")
+      XCTAssertEqual(value.project.state, .live)
+
+      XCTAssertEqual(value.backing?.status, .pledged)
+      XCTAssertEqual(value.backing?.errorReason, "This just isn't your day.")
+      XCTAssertEqual(value.backing?.pledgedOn, 1_587_502_131)
+      XCTAssertEqual(value.backing?.amount, Money(amount: "146.0", currency: .usd, symbol: "$"))
+
+      XCTAssertEqual(value.backing?.bankAccount?.bankName, "Chase")
+      XCTAssertEqual(value.backing?.bankAccount?.id, "60922339")
+      XCTAssertEqual(value.backing?.bankAccount?.lastFour, "1234")
+
+      XCTAssertEqual(value.backing?.creditCard?.id, "60981339")
+      XCTAssertEqual(value.backing?.creditCard?.lastFour, "1234")
+      XCTAssertEqual(value.backing?.creditCard?.expirationDate, "2023-04-01")
+      XCTAssertEqual(value.backing?.creditCard?.paymentType, .creditCard)
+      XCTAssertEqual(value.backing?.creditCard?.type, .visa)
+
+      XCTAssertEqual(value.backing?.shippingAmount, Money(amount: "17.0", currency: .usd, symbol: "$"))
+
+      XCTAssertEqual(value.backing?.reward?.name, "Everyday Carry")
+      XCTAssertEqual(value.backing?.reward?.backersCount, 593)
+      XCTAssertEqual(
+        value.backing?.reward?.description,
+        "For the typist who takes their keyboard everywhere."
+      )
+      XCTAssertEqual(value.backing?.reward?.estimatedDeliveryOn, "2020-08-01")
+      XCTAssertEqual(value.backing?.reward?.items?[0].id, "UmV3YXJkSXRlbS03OTczNTM=")
+      XCTAssertEqual(
+        value.backing?.reward?.items?[0].name,
+        "Keyboardio Atreus (Choose switches after campaign)"
+      )
+      XCTAssertEqual(value.backing?.reward?.items?[1].id, "UmV3YXJkSXRlbS04NzMzMDY=")
+      XCTAssertEqual(value.backing?.reward?.items?[1].name, "Travel case")
+
+      XCTAssertEqual(value.backing?.reward?.amount, Money(amount: "129.0", currency: .usd, symbol: "$"))
+
+      XCTAssertEqual(value.backing?.backer?.id, "VXNlci0xMTA4OTI0NjQw")
+      XCTAssertEqual(value.backing?.backer?.name, "Backer McGee")
+    } catch {
+      XCTFail((error as NSError).description)
+    }
+  }
+
+  func testJSONParsing_WithPartialData() {
+    let dictionary: [String: Any] = [
+      "project": [
+        "projectSummary": [
+          [
+            "question": "WHAT_IS_THE_PROJECT",
+            "response": "A cool project."
+          ],
+          [
+            "question": "WHO_ARE_YOU",
+            "response": "I am a writer."
+          ]
+        ]
+      ]
+    ]
+
+    guard let data = try? JSONSerialization.data(withJSONObject: dictionary, options: []) else {
+      XCTFail("Should have data")
+      return
+    }
+
+    let value = try? JSONDecoder().decode(ProjectSummaryEnvelope.self, from: data)
+
+    XCTAssertNotNil(value, "Should deserialize with only some values.")
+  }
+
+  func testJSONParsing_WithUnknownQuestion() {
```

Comment:
Unknown question? Que?

Reply:
This is supposed to be `testJSONParsing_NoErrorReason` I think, another copy-paste error.